carbon-design-system/carbon-addons-iot-react
# feat(progressindicator): extend component features

Closes #2605

**Summary**

- The optional `secondaryLabel` of ProgressIndicator step is displayed with ellipsis and native html tooltip on-hover when long text is provided, similar to step's label text.
- New boolean property `spaceEqually` is added, when set to `true` it makes ProgressIndicator steps to extend/adjust and spaced equally within available flex container.
- `spaceEqually` is exposed as a boolean propTypes in `PageWizard` component which uses ProgressIndicator to allow steps to be set space equally.

**Change List (commits, features, bugs, etc)**

- Altered `width` of optional label text to `width: auto` to enable ellipsis and native tooltip on-hover for lengthy text.
- Added optional `spaceEqually` prop to `ProgressIndicator` component. When set to `true` it will overwrite the `stepWidth` prop and add `flex-grow: 1` to all `li` elements(or steps) hence extending and spacing equally them within available flex container. This prop is needed in cases where there are fewer steps to display(like 2 or 3) and should be responsive with screen size.
- Added 'Steps spaced equally' story under `ProgressIndicator`.
- Added 'With steps spaced equally' story under `PageWizard`.
- Added unit test for `spaceEqually` prop `true` case in `ProgressIndicator` component.

**Acceptance Test (how to verify the PR)**

- Go to /?path=/story/1-watson-iot-progressindicator--presentation and ensure that long secondary step labels are displayed with ellipsis and tooltip with full text is displayed on-hover.
- Check the spaceEqually option under knobs section. The progress indicator steps should extend/adjust to cover the flex container and should be equally spaced. Resize the screen to check the progress indicator is responsive.
- Ensure the stepWidth option is not displayed when spaceEqually option is checked.
- Go to /?path=/story/1-watson-iot-pagewizard--with-sticky-footer-stateful-example-w-validation-in-page-title-bar and spaceEqually option is checked. Steps should adjust to the container width and should be equally spaced. Resize the screen to check it is responsive.
- Use the RTL knob to ensure everything adjusts correctly.

**Regression Test (how to make sure this PR doesn't break old functionality)**

- All remaining stories under ProgressIndicator and PageWizard works as before. Any other component story using ProgressIndicator should work as before.
- Go to /?path=/story/1-watson-iot-progressindicator--presentation ensure stepWidth prop should work as before when spaceEqually is not checked. 

https://deploy-preview-2622--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-progressindicator--presentation
https://deploy-preview-2622--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagewizard--with-sticky-footer-stateful-example-w-validation-in-page-title-bar



**Things to look for during review**
